### PR TITLE
Prometheus instrumentation

### PR DESCRIPTION
**Pull request by danwilliams**
## Changes

Replicated the instrumentation changes from the `govuk-coronavirus-vulnerable-people-form` repo here, such that request level metrics are served on the `metrics` URL. 

## How to review

A test has been added to check that the `metrics` endpoint returns a 200, this test can be run (along with the other tests in the repo) using the usual command.

```bash
bundle exec rake
```

In addition to performing a manual test one can run the server via:

```bash
foreman start
```

And go to [http://localhost/metrics](http://localhost/metrics) and see that metrics are being returned. 

**Note**: Metrics are gathered as requests are made, so on the first visit it might look somewhat empty. If you refresh the page you'll see the metrics for the first load displayed... if you visit other URLs on the app, on subsequent visits to the metrics page you'll see data for those URLs displayed too.

## Links

https://trello.com/c/E162DRPh/409-instrument-business-app

### CODE REVIEW

**Comment by huwd** [comment]
Thanks for that, "the how to review" section was especially useful, hadn't tried to spin up prometheus locally before:

Confirmed it's working
image

Question... @bilbof on the other app is `/metrics` beind basic auth?
We'd want to secure this one too right?

**Review by huwd — COMMENTED** [review]
Code looks great,
Would like to hear back from Bill/Issy if we need to put this behind basic auth too.

**Review by bilbof — DISMISSED** [review]
Looks good! Just have the [same comment as here](https://github.com/alphagov/govuk-coronavirus-find-support/pull/195#pullrequestreview-411655143) that we'll need to do those cf commands to complete these tasks.

File: concourse/tasks/deploy-to-govuk-paas.yml [hunk 1]
```diff
@@ -20,8 +20,6 @@ params:
   INSTANCES:
   WORKER_INSTANCES:
   BASIC_AUTH_PASSWORD:
-  METRICS_USERNAME: ((metrics-username))
-  METRICS_PASSWORD: ((metrics-password))
```
Comment (bilbof) [inline concourse/tasks/deploy-to-govuk-paas.yml]: Would you be able to squash this commit into the previous one before merging?

File: concourse/tasks/deploy-to-govuk-paas.yml [hunk 2]
```diff
@@ -59,3 +59,8 @@ run:
       cf set-env govuk-coronavirus-business-volunteer-form PAAS_ENV "$CF_SPACE"
       cf v3-zdt-push govuk-coronavirus-business-volunteer-form --wait-for-deploy-complete --no-route
       cf map-route govuk-coronavirus-business-volunteer-form cloudapps.digital --hostname "$HOSTNAME"
+
+      cf map-route govuk-coronavirus-business-volunteer-form coronavirus-business-volunteer-form.service.gov.uk --path metrics
```
Comment (bilbof) [inline concourse/tasks/deploy-to-govuk-paas.yml]:
```suggestion
      cf map-route govuk-coronavirus-business-volunteer-form coronavirus-business-volunteers.service.gov.uk --path metrics
```

File: concourse/tasks/deploy-to-govuk-paas.yml [hunk 3]
```diff
@@ -59,3 +59,8 @@ run:
...
+      cf map-route govuk-coronavirus-business-volunteer-form cloudapps.digital --hostname govuk-coronavirus-business-volunteer-form-prod --path metrics
+      cf bind-route-service coronavirus-business-volunteer-form.service.gov.uk re-ip-whitelist-service --path metrics
```
Comment (bilbof) [inline concourse/tasks/deploy-to-govuk-paas.yml]:
```suggestion
      cf bind-route-service coronavirus-business-volunteers.service.gov.uk re-ip-whitelist-service --path metrics
```

**Review by bilbof — COMMENTED** [review]
Looks good. Nice work inferring the `cf` stuff :) just a couple of changes requested

**Review by bilbof — APPROVED** [review]

**Pull request merged**